File: nodes/get-messages-by-role/get-messages-by-role.process.py
from typing import Any, Dict, List, Union

async def process(inputs: Dict[str, Any], settings: Dict[str, Any], config: Dict[str, Any], nodeConfig: Dict[str, Any]) -> Dict[str, Any]:
    try:
        messages = inputs.get('messages')
        role = inputs.get('role')

        if not isinstance(messages, list):
            raise Exception("Messages input must be an array")

        if not isinstance(role, str) or not role.strip():
            raise Exception("Role must be a non-empty string")

        # Filter messages by role
        filtered_messages = [
            message for message in messages 
            if isinstance(message, dict) and message.get('role') == role
        ]

        return {
            "messages": filtered_messages,
            "count": len(filtered_messages)
        }

    except Exception as error:
        print(f'Get Messages by Role error: {error}')
        raise Exception(f"Get Messages by Role error: {str(error)}")

File: sdks/nodejs/nodes/get-messages-by-role/get-messages-by-role.process.js
export default async ({inputs, settings, config}) => {
    try {
        const messages = inputs.messages;
        const role = inputs.role;

        if (!Array.isArray(messages)) {
            throw new Error("Messages input must be an array");
        }

        if (typeof role !== 'string' || role.trim() === '') {
            throw new Error("Role must be a non-empty string");
        }

        // Filter messages by role
        const filteredMessages = messages.filter(message => {
            return message && typeof message === 'object' && message.role === role;
        });

        return {
            messages: filteredMessages,
            count: filteredMessages.length
        };

    } catch (error) {
        console.log('Get Messages by Role error:', error);
        throw new Error(`Get Messages by Role error: ${error.message}`);
    }
};
